test-cypress/integration/sparql/main.menu.spec.js

```diff
@@ -2,186 +2,185 @@ describe('Main menu tests', function () {
+                        redirect: '/repository'
+                    },
+                    {
+                        name: 'Users and Access',
+                        visible: false,
+                        redirect: '/users'
+                    },
+                    {
+                        name: 'My Settings',
+                        visible: false,
+                        redirect: '/settings'
+                    },
+                    {
+                        name: 'Connectors',
+                        visible: false,
+                        redirect: '/connectors'
+                    },
+                    {
+                        name: 'Namespaces',
+                        visible: false,
+                        redirect: '/namespaces'
+                    },
+                    {
+                        name: 'Autocomplete',
+                        visible: false,
+                        redirect: '/autocomplete'
+                    },
+                    {
+                        name: 'RDF Rank',
+                        visible: false,
+                        redirect: '/rdfrank'
+                    }
+                ]
+            },
+            {
+                name: 'Help',
+                visible: true,
+                submenu: [
+                    {
+                        name: 'REST API',
+                        visible: false,
+                        redirect: '/webapi'
+                    },
+                    {
+                        name: 'Documentation',
+                        visible: false,
+                        externalRedirect: 'http://graphdb.ontotext.com/documentation/'
+                    },
+                    {
+                        name: 'Developer Hub',
+                        visible: false,
+                        externalRedirect: '/devhub/'
+                    },
+                    {
+                        name: 'Support',
+                        visible: false,
+                        externalRedirect: 'http://graphdb.ontotext.com/'
+                    },
+                    {
+                        name: 'System information',
+                        visible: false,
+                        redirect: '/sysinfo'
+                    }
+                ]
+            }
+        ];
+
+        menuItems.forEach((menu, menuIndex) => {
+            let menuVisibilityCheck = menu.visible ? 'be.visible' : 'not.be.visible';
+            // Verify that main menu items are present and their visibility
+            cy.get('.main-menu .menu-element-root').eq(menuIndex + 1).as('menu')
+                .should(menuVisibilityCheck).and('contain', menu.name);
+
+            // Verify submenu items and their visibility when the main menu is not opened
+            (menu.submenu || []).forEach((submenu, submenuIndex) => {
+                let submenuVisibilityCheck = submenu.visible ? 'be.visible' : 'not.be.visible';
```

Comment:
All sub menu items are not visible until opened, right? I don't see the need of ` visible: false` in `menuItems`

Reply:
That's right, but I decided to have it in for completeness. If you insist I can remove it thou